### PR TITLE
FIX# Displaying 'Diff' in new line

### PR DESCRIPTION
**Pull request by sivagollapalli**
Issue #875

### CODE REVIEW

**Comment by JonRowe** [comment]
Thanks for tackling this, could you add a spec for it?

File: lib/rspec/mocks/error_generator.rb [hunk 1]
```diff
@@ -65,7 +65,7 @@ def raise_missing_default_stub_error(expectation, *args)
         diff = diff_message(expectation.expected_args, args)
 
         message = default_error_message(expectation, expected_args, actual_args)
-        message << "Diff:\n #{diff}" unless diff.empty?
+        message << "\n Diff:\n #{diff}" unless diff.empty?
```
Comment (myronmarston) [inline lib/rspec/mocks/error_generator.rb]: In `raise_unexpected_message_args_error`, we use `"\nDiff"`, which left-aligns `Diff` against the margin, whereas this has Diff indented one space.  I don't think we want that space.

**Comment by imtayadeway** [comment]
I began looking at https://github.com/rspec/rspec-mocks/issues/875 before realizing there was already this open PR for it. I decided to complete it and address the test coverage as @JonRowe suggests. Not to wish to interfere here, but I am happy to help try to close this if @sivagollapalli is unable to. Thanks!

**Comment by sivagollapalli** [comment]
@JonRowe Added spec for the issue. @myronmarston Removed extra space as you suggested.

**Comment by myronmarston** [comment]
Thanks, @sivagollapalli!

**Pull request merged**